features/ubuntu_pro.feature

```diff
@@ -660,3 +660,47 @@ Feature: Command behaviour when auto-attached in an ubuntu PRO image
            | bionic  |
            | focal   |
            | jammy   |
+
+    @series.xenial
+    @series.jammy
+    @uses.config.machine_type.gcp.pro
+    Scenario Outline: Unregistered Pro machine
+        Given a `<release>` machine with ubuntu-advantage-tools installed
+        When I create the file `/etc/ubuntu-advantage/uaclient.conf` with the following:
+        """
+        contract_url: 'https://contracts.canonical.com'
+        data_dir: /var/lib/ubuntu-advantage
+        log_level: debug
+        log_file: /var/log/ubuntu-advantage.log
+        """
+        And I create the file `/tmp/response-overlay.json` with the following:
+        """
+        {
+            "https://contracts.canonical.com/v1/clouds/gcp/token": [
+            {
+              "code": 400,
+              "type": "contract",
+              "response": {
+                "code": "contract not found",
+                "message": "missing product mapping for subscription \\"subscription\\", plan \\"plan\\", product \\"product\\", publisher \\"canonical\\", sku \\"sku\\" ",
+                "traceId": "id"
+              }
```

Comment:
@lucasmoura This is what the response looks like on azure.

On aws it looks largely the same:
```
{'code': 'contract not found', 'message': 'missing instance information', 'traceId': '$TRACEID'}
```

On gcp it looks a bit different:
```
{'code': 'internal server error', 'message': 'missing product mapping for licenses [$LICENSEID]', 'traceId': '$TRACEID'}
```

So I think we may want to make this PR a bit more generic. Maybe we can just directly log/print the code and message that the server gives us?

Reply:
Good point, let's not tie this to the error code.

Reply:
@orndorffgrant Code update to be more generic